cmd/acr/purge_test.go

```diff
@@ -559,6 +559,70 @@ func TestDryRun(t *testing.T) {
 	})
 }
 
+// TestCollectTagFilters contains all the tests regarding the collectTagFilters with retrieves matching repo names
+// and aggregates the associated tag filters
+func TestCollectTagFilters(t *testing.T) {
+	t.Run("SingleRepo", func(t *testing.T) {
+		assert := assert.New(t)
+		mockClient := &mocks.BaseClientAPI{}
+		mockClient.On("GetRepositories", mock.Anything, "", (*int32)(nil)).Return(ManyRepositoriesResult, nil).Once()
+		filters, err := collectTagFilters(testCtx, []string{testRepo + ":.*"}, mockClient)
+		assert.Equal(1, len(filters), "Number of found should be one")
+		assert.Equal(".*", filters[testRepo], "Filter for test repo should be .*")
+		assert.Equal(nil, err, "Error should be nil")
+		mockClient.AssertExpectations(t)
+	})
+	t.Run("AllRepos", func(t *testing.T) {
+		assert := assert.New(t)
+		mockClient := &mocks.BaseClientAPI{}
+		mockClient.On("GetRepositories", mock.Anything, "", (*int32)(nil)).Return(ManyRepositoriesResult, nil).Once()
+		filters, err := collectTagFilters(testCtx, []string{":.*"}, mockClient)
+		assert.Equal(4, len(filters), "Number of found should be 4")
+		assert.Equal(".*", filters[testRepo], "Filter for test repo should be .*")
+		assert.Equal(".*", filters["bar"], "Filter for bar repo should be .*")
+		assert.Equal(nil, err, "Error should be nil")
+		mockClient.AssertExpectations(t)
+	})
+	t.Run("AllReposWildcard", func(t *testing.T) {
+		assert := assert.New(t)
+		mockClient := &mocks.BaseClientAPI{}
+		mockClient.On("GetRepositories", mock.Anything, "", (*int32)(nil)).Return(ManyRepositoriesResult, nil).Once()
+		filters, err := collectTagFilters(testCtx, []string{":.*"}, mockClient)
```

Comment:
What's the difference between this case and the last one? Do you mean `".*:.*"` here?

Reply:
Related to your comment below:
https://github.com/Azure/acr-cli/pull/88#discussion_r804442611

Please decide if this is desirable or not

Reply:
Fixed